consensus/core/src/dag_state.rs

```diff
@@ -351,20 +483,43 @@ impl DagState {
         self.last_commit = Some(commit);
     }
 
+    pub(crate) fn genesis_blocks(&self) -> Vec<VerifiedBlock> {
+        self.genesis.values().cloned().collect()
+    }
+
     /// Highest round where a block is committed, which is last commit's leader round.
     fn last_commit_round(&self) -> Round {
         match &self.last_commit {
             Some(commit) => commit.leader.round,
             None => 0,
         }
     }
+
+    /// Should be used only for testing. Retains only the cache elements that are `< round` and flushes
+    /// the rest in store.
+    #[cfg(test)]
+    fn flush_cache_for_testing(&mut self, round: Round) {
```

Comment:
This is not a full flush, since commit and other values are not written. Maybe just `write_blocks_below_round()`.